### DISCUSSION TITLE
Third party variants in this repo?

### DISCUSSION THREAD

**Discussion by owennewo** [stm32duino/Arduino_Core_STM32#1297] [General]
What is your policy of accepting popular third party boards (not built by STM) that have stm32 chips on them?

As an example I'd be willing to put together a board variant for the [open source VESC 4.12 board](http://vedder.se/2015/01/vesc-open-source-esc/) designed by Benjamin Vedder which is a stm32f405 board with a powerful driver/mosfet for BLDC motor?

**Comment by fpistm** [comment]
Hi @owennewo 
If you look at the README.md you will see there are already several boards not manufactured by ST. The STM32 core is for all boards based on a STM32 MCU.
The board you talk about could be added to the generic L4 variant which will be available in the coming weeks as I fully rework the way to define a variant to ease further boards addition. (#1091)

**Reply by owennewo** [reply]
@fpistm - when will #1091 be completed? I'm ready to start working on a vesc board variant - but will wait if your refactor is around the corner

**Reply by fpistm** [reply]
In the coming weeks...

**Reply by owennewo** [reply]
Is #1091 on a branch that I can look at?  I'd like to try to get my new variant ready using the new variant format

**Reply by fpistm** [reply]
#1091 is always in draft. At the end of the week I hope to moved as ready for review.

**Comment by owennewo** [comment]
I have the VESC 4.12 board variant on my forked repo but will hold from doing PR until after Milestone 2.0
https://github.com/owennewo/Arduino_Core_STM32/tree/vesc_4_12
https://github.com/owennewo/platform-ststm32/tree/vesc_4_12

The board is working pretty well - I've tested CAN, USART6, SPI1, I2C2, SerialUSB.
With SimpleFOC library I've got 6PWM closed loop motor control working.

Hopefully 2.0 won't be that long
